### ISSUE TITLE
Ubuntu PPA

### ISSUE THREAD

**Issue by OakBricks** [johanmalm/jgmenu#126]
I'm not too sure if this is a good request but, basically make a ubuntu PPA for ubuntu users so that they don't need to use make or dpkg-buildpackage -tc -b -us -uc. This is mostly for ease of use

**Comment by johanmalm** [comment]
Hi and thanks for writing. I’m trying to separate myself from packaging, because it requires different skills and is time consuming. I have tried before (including Ubunutus PPA), but regretted it.
Maybe something you’d want to have a go at yourself? The debian/ folder is quite tidy. Happy to help/advice, but don’t want to lead on it.

**Comment by alexmyczko** [comment]
https://mentors.debian.net/package/jgmenu/

**Comment by johanmalm** [comment]
@alexmyczko Thanks
I've fixed the typos.
What is the intent of uploading it to 'https://mentors.debian.net/package/jgmenu/'? Is it a route to make it an official debian package?
Is there anything I should be doing?

**Comment by OakBricks** [comment]
Yeah about your first comment i've had some issues making a .changes file for jgmenu seems to be an issue with one of the files on the "debian" folder, i don't know a lot about PPA's and less about making the .changes file, it seems to produce a .changes file, but I know nothing on GPG signing, the good thing is i've went far enough to make a (potentially unusable/unstable) .changes, so i'll try uploading it to launchpad when I get the time (Friday maybe)

**Comment by alexmyczko** [comment]
> @alexmyczko Thanks
> I've fixed the typos.

Thanks!

> What is the intent of uploading it to 'https://mentors.debian.net/package/jgmenu/'? 

Leandro Cunha <leandrocunha016@gmail.com> did that, I don't know who that is.
But it's a place for everyone to suggest a package to be sponsored by an official Debian Developer to review the packaging
and possibly upload it to Debian.

> Is it a route to make it an official debian package?

Yes. (And Ubuntu autosyncs their packages from Debian, if that happens the PPA becomes obsolete)

> Is there anything I should be doing?

At the moment I don't think so :)

**Comment by alexmyczko** [comment]
Hi,

> Yeah about your first comment i've had some issues making a .changes file for jgmenu seems to be an issue with one of the files on the "debian" folder, i don't know a lot about PPA's and less about making the .changes file, it seems to produce a .changes file

I've never written a .changes file manually, it gets created with `debuild` for me.

> but I know nothing on GPG signing, the good thing is i've went far enough to make a (potentially unusable/unstable) .changes, so i'll try uploading it to launchpad when I get the time (Friday maybe)

If you have a GPG (PGP) key, see `man gpg`, you can sign your package. I guess you use `dput`? Uploading stuff to PPA is not too difficult.

**Comment by OakBricks** [comment]
@alexmyczko 
1. I meant using debuild
2. Thanks, and yes I do use dput for PPAs

**Comment by Head-on-a-Stick** [comment]
Ubuntu packages here:

https://software.opensuse.org//download.html?project=home%3AHead_on_a_Stick%3Ajgmenu&package=jgmenu

It's an openSUSE Build Service repository but it functions in the same way as a PPA. Either download and install the .debs or add the repository manually, `add-apt-repository` doesn't work for the OBS.

**Comment by OakBricks** [comment]
Its a good step but I don;t want to open up nano or leafpad to edit my sources.list, I just want to to do apt-add-repository ppa:USERNAME/REPO

**Comment by Head-on-a-Stick** [comment]
> I don't want to open up nano or leafpad to edit my sources.list,

It is recommended to use `apt edit-sources` to edit `/etc/apt/sources.list` directly, that will perform a syntax check during the save and malformed lines will be rejected.

To add the repository without using an editor copy & paste the lines given in the instructions on the packages page I linked in my last post.

Sorry but I'm not prepared to interact with Comical's PPA infrastructure and I certainly won't install Ubuntu for testing, not even in a VM :-)
